### DISCUSSION TITLE
@SpringBootTest vs @DataJpaTest | Devdalus Wiki

### DISCUSSION THREAD

**Discussion by giscus** [DevdaIus/DevdaIus.github.io#3] [General]
# @SpringBootTest vs @DataJpaTest | Devdalus Wiki

Tech Archive for project Devdalus

https://devdaius.github.io/docs/test/spring_boot_test_vs_data_jpa_test.html

**Comment by claycat** [comment]
> @SpringBootTest + @AutoConfigureTestDatabase + @Transactional을 조합해서 사용하면 @DataJpaTest와 동일하게 동작하게 할 수 있다.

@DataJpaTest 는 컴포넌트들을 등록시켜주지는 않는걸로 알고있어요
그래서 오히려 DataJpaTest가 SpringBootTest보다 더 범위가 작다고 생각하는데 아닌가요?

**Reply by punkryn** [reply]
맞습니다 제가 잘못적었네요
동일하게 동작한다 ->embedded db와 transaction을 사용할 수 있다로 수정

**Comment by claycat** [comment]
@DataJpaTest 애노테이션 선언을 가보면 실제로 @Transactional이 들어있네요!
